docs/src/2.5.AddingInteractivity.mdx

```diff
@@ -0,0 +1,116 @@
+import CodeSandboxEmbed from './jsx/utils/CodeSandboxEmbed.js';
+
+# Adding Interactivity
+
+In this step, you'll learn how to:
+
+- Set up the marker input for hitmap/mouse interaction
+- Interact with the scene through event handlers
+- Retrieve object and event information from the event handlers
+
+## Adding and Removing Obstacles along the Duck's Path
+
+In previous steps, you've learned how to interact with Worldview through mouse, keyboard, and programatic controls, all of which affect the camera. Worldiview and its Commmand components also support other [Mouse Events](#/docs/api/mouse-events), such `onClick`, `onDoubleClick`, `onMouseDown`, `onMouseMove`, and `onMouseUp`.
+
+In this step, let's place red cubes as obstacles along the duck's path. When the duck hits an obstacle, it should stop moving until you removed the blocking obstacle.
+
+First, let's add an `id` field to the sphere marker so that we can identify the sphere being clicked.
+
+```js
+const marker = {
+  // hitmap support requires unique positive integer id
+  id: 101,
+  pose: {
+    orientation: { x: 0, y: 0, z: 0, w: 1 },
+    position: { x: 0, y: 0, z: 0 },
+  },
+  scale: { x: 1, y: 1, z: 1 },
+  colors: [],
+  points: [],
+};
+```
+
+Let's also enable manual camera movement, so we can see the duck better.
+
+```js
+<Worldview defaultCameraState={{ ...DEFAULT_CAMERA_STATE, distance: 120 }}>
+```
+
+Next, let's add a `clickedObjectIds` state to keep track of the clicked spheres. We'll also use this `clickedObjectIds` state to generate cube obstacles.
+
+```js
+const [clickedObjectIds, setClickedObjectIds] = useState(new Set());
+
+const obstacleMarkers = Array.from(clickedObjectIds).map((clickedObjectId, index) => {
+  const pointIdx = clickedObjectId - marker.id;
+  const position = marker.points[pointIdx];
+  return {
+    // keep the id unique by adding sphere's id range as offset: marker.id + steps
+    id: marker.id + steps + index,
+    // remember the original clickedObjectId so when the obstacle is clicked, we can
+    // remove the obstacle quickly by updating clickedObjectIds
+    clickedObjectId,
+    pose: {
+      orientation: { x: 0, y: 0, z: 0, w: 1 },
+      position: position,
+    },
+    color: { r: 1, g: 0, b: 0, a: 1 }, // red
+    scale: { x: 4, y: 4, z: 4 }, // scale up a little so it's bigger than the spheres
+  };
+});
+```
+
+Now, let's use the clicked sphere id to update the `clickedObjectIds` state, which will be consumed by the cube obstacles. Clicking each obstacle should remove it from the scene.
+
+```js
+ <Spheres
+    onClick={(ev, { objectId, object }) => {
+        // click on the object to add an obstacle
+        clickedObjectIds.add(objectId);
+        setClickedObjectIds(new Set(clickedObjectIds));
```

Comment:
Seems pointless to use `.add()` and then `new Set()`. At least we could use `setClickedObjectIds(clickedObjectIds.add(objectId))` (assuming that will actually cause a re-render, which I think it will). Alternatively, could just use an array and then `setClickedObjectIds([...clickedObjectIds, objectId])` for full immutability (although it's a bit less efficient, but that doesn't matter much because the number of objects will be quite small)